### PR TITLE
Current angle based direction reversal

### PR DESCRIPTION
**Pull request by PhilipAxelrod**
I could use some help reducing wobble in the last 2 integration tests of pure pursuit. The general path followed seems correct, but there is an oscillation in the angle

### CODE REVIEW

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/MathUtilities.scala [hunk 1]
```diff
@@ -99,19 +99,22 @@ object MathUtilities {
...
+    * (right is 0 degrees, increasing counter clockwise) to
+    * compass angle, forward is 0, increases clockwise, or vice versa
+    * @param trigonemtricAngle
+    * @return
+    */
+  def swapTrigonemtricAndCompass(trigonemtricAngle: Angle): Angle = {
+    Degrees(90) - trigonemtricAngle
+  }
+
+  def clamp[T](toClamp: Dimensionless, max: Dimensionless): Dimensionless = {
```
Comment (shadaj) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/MathUtilities.scala:113]: At some point we could probably add this as a feature to Squants! `quantity.clamp(range)` or something like that for more generic support of clamping.

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala [hunk 2]
```diff
@@ -1,12 +1,12 @@
 package com.lynbrookrobotics.potassium.commons.drivetrain
 
+import java.io.{File, PrintWriter}
```
Comment (shadaj) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala]: Uhhh this concerns me are we writing to files in our control loops?

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala [hunk 3]
```diff
@@ -201,14 +199,12 @@ trait PurePursuitControllers extends UnicycleCoreControllers {
       }
     }
 
-    (limitedForward.zip(turnOutput).zip(multiplier).zip(lookAheadPoint).zip(forwardError).map { o =>
-      val ((((forward, turn), fdMultiplier), la), frdError) = o
-      if (frdError > props.get.defaultLookAheadDistance / 2) {
+    (
```
Comment (shadaj) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala:200]: Ooh this is so much simpler 🎉

File: model/src/main/scala/com/lynbrookrobotics/potassium/model/simulations/SimulatedTwoSidedHardware.scala [hunk 4]
```diff
@@ -112,22 +129,31 @@ class SimulatedTwoSidedHardware(constantFriction: Force,
 
   val position = XYPosition(turnPosition.map(a => Degrees(90) - a), forwardPosition)
 
-  val zippedPositions = forwardPosition.zip(turnPosition).zip(position)
-
-  val historyStream = zippedPositions.zip(forwardVelocity).zip(turnVelocity).zipWithTime.map {
-    case (((((fPos, tPos), pos), fVel), tVel), time) =>
-      MomentInHistory(
-        time = time,
+  val zippedStuff: Stream[(((((((Length, Angle), Point), Velocity), AngularVelocity), Dimensionless), Dimensionless), Time)] = forwardPosition.zip(turnPosition)
```
Comment (shadaj) [inline model/src/main/scala/com/lynbrookrobotics/potassium/model/simulations/SimulatedTwoSidedHardware.scala]: Maybe not include the explicit type? I feel that makes it look a lot scarier than it actually is. Your call though.

File: model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/SimulatePurePursuit.scala [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+import squants.motion._
+import squants.space._
+
+import scala.collection.immutable.{NumericRange, Range}
+import com.lynbrookrobotics.potassium.units._
+import com.lynbrookrobotics.potassium.units.GenericValue._
+
+import scala.reflect.io.File
+
+class SimulatePurePursuit extends FunSuite {
```
Comment (shadaj) [inline model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/SimulatePurePursuit.scala:25]: YAYY I ❤️  this 😍

**Review by shadaj — CHANGES_REQUESTED** [review]
One change to remove extra imports and one question, but otherwise LGTM!

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala (hunk 2 above)
Comment (PhilipAxelrod) [reply commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala]: Oh, oops. I was logging stuff, then I removed the logging.

File: model/src/main/scala/com/lynbrookrobotics/potassium/model/simulations/SimulatedTwoSidedHardware.scala [hunk 6]
```diff
@@ -129,7 +129,7 @@ class SimulatedTwoSidedHardware(constantFriction: Force,
 
   val position = XYPosition(turnPosition.map(a => Degrees(90) - a), forwardPosition)
 
-  val zippedStuff: Stream[(((((((Length, Angle), Point), Velocity), AngularVelocity), Dimensionless), Dimensionless), Time)] = forwardPosition.zip(turnPosition)
+  val zippedStuff = forwardPosition.zip(turnPosition)
```
Comment (shadaj) [inline model/src/main/scala/com/lynbrookrobotics/potassium/model/simulations/SimulatedTwoSidedHardware.scala]: Maybe call this `robotStateStream` or something more descriptive?

**Review by shadaj — APPROVED** [review]

**Pull request merged**